src/linked_hash_map.rs

```diff
@@ -71,6 +72,22 @@ impl<K: PartialEq> PartialEq for KeyRef<K> {
 
 impl<K: Eq> Eq for KeyRef<K> {}
 
+// This type exists only to support borrowing `KeyRef`s, which cannot be borrowed to `Q` directly
+// due to conflicting implementations of `Borrow`. The layout of `&Qey<Q>` must be identical to
+// `&Q` in order to support transmuting in the `Qey::from_ref` method.
```

Comment:
Agreed, but this entire module is pretty hacky. Presumably a safer version of this collection could be written if it had access to the internal representation of `HashMap`.